libcyphal_demo/src/platform/string.hpp

```diff
@@ -0,0 +1,334 @@
+// This software is distributed under the terms of the MIT License.
```

Comment:
Do you think there's any point in getting this into CETL?

Reply:
I'm not sure that there is... Hypothetically, if it would be helpful for Nunavut c++ generation then probably yes, but I don't see it currently needed there. Except maybe for some "string"-related Nunavut types (like `String_1_0`, `register::Name_1_0` etc) to make them dynamic memory allocation free - still not sure it's good idea b/c it will affect (footprint) composite types (like `Value_1_0`).